README.md

```diff
@@ -442,23 +444,78 @@ We have a small collection of algorithms, one for every day of the month. Scroll
     - **Hint:** Try something with the ASCII values of each word maybe.
-    
+
   - **_Optional Problem_**
     - Hash functions may return the same output for various inputs. This is known as collision. There are two main collision resolution techniques:
       - [Separate Chaining](https://youtu.be/_xA8UvfOGgU)
       - [Open Addressing](https://youtu.be/Dk57JonwKNk)
     - Implement a hash table with one of the collision resolution techniques mentioned above.
-    
+
   - **Uses**
     - As mentioned above, hash functions are often used in combination with a hash table for rapid data lookup.
     - They are also useful in cryptography. A cryptographic hash function allows one to easily verify that some input data maps to a given hash value, but if the input data is unknown, it is deliberately difficult to reconstruct it (or any equivalent alternatives) by knowing the stored hash value.
     - Cryptographic hash functions also play a crucial role in blockchains.
-    
+
   - **Resources**
     - [Hash function – Wikipedia](https://en.wikipedia.org/wiki/Hash_function)
     - [Hash table - Wikipedia](https://en.wikipedia.org/wiki/Hash_table)
     - [An in-depth video about cryptocurrencies, blockchain and hash functions](https://youtu.be/bBC-nXj3Ng4)
-    
+
+
+### **December 20 - Dijkstra's Shortest Reach**
+  - **Problem**
+    - Given an undirected graph and a starting node, determine the lengths of the shortest paths from the starting node to all other nodes in the graph. If a node is unreachable, its distance is `-1` or `INFINITE`. Nodes will be numbered consecutively from to, and edges will have varying distances or lengths.
+    For example, consider the following graph of 9 nodes:
+
+     <img src="/src/docs/Dec20-Example.jpg" width="550" height="250"/>
+
+    - Write an algorithm that returns an array of integers that represent the shortest distance to each node from the start node in ascending order of node number, given the number of nodes, the edges and starting node number.
+
+  - **Example**
+    ```
+    {0, 4, 0, 0, 0, 0, 0, 8, 0}
+    {4, 0, 8, 0, 0, 0, 0, 11, 0}
+    {0, 8, 0, 7, 0, 4, 0, 0, 2}
+    {0, 0, 7, 0, 9, 14, 0, 0, 0}
+    {0, 0, 0, 9, 0, 10, 0, 0, 0}
+    {0, 0, 4, 14, 10, 0, 2, 0, 0}
+    {0, 0, 0, 0, 0, 2, 0, 1, 6}
+    {8, 11, 0, 0, 0, 0, 1, 0, 7}
+    {0, 0, 2, 0, 0, 0, 6, 7, 0}
+    ```
+    ```
+    Vertex   Distance from Source
+    0                0
+    1                4
+    2                12
+    3                19
+    4                21
+    5                11
+    6                9
+    7                8
+    8                14
+    ```
+  - **Resources**  
```

Comment:
We're gonna need way more resources :P

I think this is the hardest program in the list so far. Need an explanation of what a graph is, what the notation in the example means, maybe a visual explanation of how Dijkstra's Algorithm works [https://visualgo.net/en/sssp or https://www.cs.usfca.edu/~galles/visualization/Dijkstra.html]

Reply:
Cool Resources! I'll update them along with a detailed explanation of terminologies involved.